src/olympia/files/models.py

```diff
@@ -378,40 +378,53 @@ def get_localepicker(self):
     def webext_permissions(self):
         """Return permissions that should be displayed, with descriptions, in
         defined order:
-        1) match all urls (e.g. <all-urls>)
-        2) known permissions, in constants order (alphabetically),
-        3) match urls for sites
+        1) Either the match all permission, if present (e.g. <all-urls>), or
+           match urls for sites (<all-urls> takes preference over match urls)
+        2) nativeMessaging permission, if present
+        3) other known permissions in alphabetical order
         """
         knowns = list(WebextPermissionDescription.objects.filter(
             name__in=self.webext_permissions_list).iterator())
-
-        match_all, urls = [], []
+        # Move nativeMessaging permission to front of the list, if present
+        for index, perm in enumerate(knowns):
```

Comment:
Could do that... but it'll take another query!  But... the first query will be cached, so would the second one actually hit the DB?
e.g. something like:
```
q = WebextPermissionDescription.objects.filter(name__in=self.webext_permissions_list)
len(q)  # Or something else to force the results to be cached
knowns = list(q.exclude(name=WebextPermission.NATIVE_MESSAGING_NAME))
...
match_url = q.get(name=WebextPermission.NATIVE_MESSAGING_NAME)
```